server/src/test/java/io/crate/integrationtests/ColumnPolicyIntegrationTest.java

```diff
@@ -182,6 +187,10 @@ public void testInsertDynamicObjectArray() throws Exception {
         assertThat(response).hasRows(
             "Ford| [West Country]"
         );
+
+        // Verify that PKLookup can fetch a sub-column, which is array of objects.
+        execute("select person['addresses']['city'] from dynamic_table where id = 1");
+        assertThat(response).hasRows("[West Country]");
     }
```

Comment:
Technically,  this scenario is covered here as in this PR we use same code for Collect/Get. 

But I think it wouldn't hurt to have at least one explicit `PKLookup` in case Collect/Get again gets diverged for whatever reason? Not likely but just to be future proof

Reply:
If it gets diverged, this code would verify that from a user perspective all is fine, or? then this is enough I'd say.